tests/Core/RouterTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+use TrendingRepos\Core\Router;
+use TrendingRepos\Core\Request;
+
+class RouterTest extends TestCase
```

Comment:
Let me analyze your test in general now. 

1. In 2 test cases out of 3, you're overwriting `$this->router` with different configs, suitable for the situation you want to test. Does it make sense to have it in `setUp()` then? 
2. I'll list below all the scenarios you came up with. Did you cover all of them?

You wanted to test when user:
* Visit a url that exist -> see the content .
* Visit a url that doesn't exist -> get the thrown exception
* Has a bad routes structure that doesn't fit mine -> get the thrown exception
* Hit a route that exist but the controller/method doesn't -> get the thrown exception

Reply:
- Removing router from setup,
- Believe me, i tried like a thousand times to figure the first test but i couldn't :disappointed:

Reply:
I decided to help you with this. Check out my commit. Had to introduce a new lightweight controller, we'll use it later when the application gain some maturity and we'll work on setting up monitoring.